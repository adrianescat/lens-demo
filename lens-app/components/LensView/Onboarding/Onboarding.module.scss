.onboarding {
  border-top: 1px solid #f0f0f0;
  position: fixed;
  padding: 38px 0;
  bottom: 0;
  background-color: #fff;
  text-align: center;
  width: calc(100% - 352px);
  left: 32px;
  opacity: 0;
  visibility: hidden;
  transition: all .6s cubic-bezier(0.15, 0.9, 0.34, 0.95);

  
  h2 {
    font-family: 'MSemiBold';
    font-size: 22px;
    color: #313131;
    margin-bottom: 15px;
    opacity: 0;
    visibility: hidden;
    transition: all .6s cubic-bezier(0.15, 0.9, 0.34, 0.95);
  }

  p {
    color: #a1a1a1;
    font-size: 16px;
    opacity: 0;
    visibility: hidden;
    text-align: center;
    max-width: 40%;
    margin: 0 auto;
    transform: translateY(21px);
    transition: all .6s cubic-bezier(0.15, 0.9, 0.34, 0.95);
  }

  &.active {
    opacity: 1;
    visibility: visible;

    h2 {
      opacity: 1;
      visibility: visible;
      transform: translate3d(0, 0, 0);
      transition: all .6s cubic-bezier(0.15, 0.9, 0.34, 0.95) 0.1s;
    }

    p {
      opacity: 1;
      visibility: visible;
      transform: translateY(0);
      transition: all .6s cubic-bezier(0.15, 0.9, 0.34, 0.95);
    }
  }
}