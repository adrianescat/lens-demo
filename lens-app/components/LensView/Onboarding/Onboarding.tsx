import React, { ReactElement } from 'react'
import classnames from 'classnames';
import styles from './Onboarding.module.scss';

interface Props {
  isActive?: boolean;
  title: string;
  description: string;
}

export default function Onboarding({
  isActive = false,
  title,
  description,
}: Props): ReactElement {
  return (
    <div className={classnames(styles.onboarding, isActive ? styles.active : '')}>
      <h3>{title}</h3>
      <p>{description}</p>
    </div>
  )
}
