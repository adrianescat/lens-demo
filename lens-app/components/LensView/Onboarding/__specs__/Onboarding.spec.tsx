import React from 'react';
import { shallow } from 'enzyme';
import Onboarding, { Props } from '../Onboarding';

let wrapper: any;
let props: Props;

beforeEach(() => {
  props = {
    title: 'title',
    description: 'description'
  }
  wrapper = shallow(<Onboarding {...props} />);
});

test('should match snapshots', () => {
  expect(wrapper).toMatchSnapshot();
});

describe('when is active', () => {
  beforeEach(() => {
    props = {
      title: 'title',
      description: 'description',
      isActive: true
    }
    wrapper = shallow(<Onboarding {...props} />);
  });

  test('should match snapshots', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
