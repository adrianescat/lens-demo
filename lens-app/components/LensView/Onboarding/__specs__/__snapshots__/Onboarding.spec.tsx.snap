// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<div
  className="onboarding"
>
  <h3>
    title
  </h3>
  <p>
    description
  </p>
</div>
`;

exports[`when is active should match snapshots 1`] = `
<div
  className="onboarding active"
>
  <h3>
    title
  </h3>
  <p>
    description
  </p>
</div>
`;
